aim/cli/up/utils.py

```diff
@@ -27,3 +32,119 @@ def build_uvicorn_command(host, port, num_workers, ssl_keyfile, ssl_certfile):
         cmd += ['--ssl-certfile', ssl_certfile]
     cmd += ['aim.web.run:app']
     return cmd
+
+
+def parse_tf_events(tf_logs, repo_inst):
+    """
+    This function scans and collects records from TF event files.
+
+    Creates and uses cache file "tf_logs_cache" in the repo dir
+    to track previously processed files and values
+
+    Currently, supports the following TF plugins
+        - scalar
+        - image
+    """
+
+    # This import statement takes long to complete
+    if not os.path.isdir(tf_logs):
+        logger.error('Value of "tf_logs" should be a valid path to the root of TF logdir.')
+        return
+
+    try:
+        import tensorflow as tf
+        from tensorflow.python.summary.summary_iterator import summary_iterator
+    except ModuleNotFoundError:
+        logger.error('Cannot process TF events - could not import required modules.')
+        return
+
+    tf_logs_cache_path = f'{repo_inst.path}/tf_logs_cache'
+    try:
+        with open(tf_logs_cache_path) as FS:
+            tf_logs_cache = json.load(FS)
+    except Exception:
+        tf_logs_cache = {}
+
+    for run_path in os.listdir(tf_logs):
+        path = os.path.join(tf_logs, run_path)
+        if path not in tf_logs_cache:
+            tf_logs_cache[path] = {}
+
+        run_cache = tf_logs_cache[path]
+        if run_cache:
+            run = Run(
+                repo=repo_inst,
+                system_tracking_interval=None,
+                run_hash=run_cache['run_hash'],
+            )
+        else:
+            run = Run(
+                repo=repo_inst,
+                system_tracking_interval=None,
+            )
+            run['tf_logdir'] = path
+            run_cache.update({
+                'run_hash': run.hash,
+                'events': {},
+            })
+        run_tf_events = run_cache['events']
+
+        events_to_process = []
+        events = [
+            os.path.join(root, file)
+            for root, dirs, files in os.walk(path)
+            for file in files if file.endswith('v2')
```

Comment:
What about `tensorboard` event files not ending with `v2`?

Reply:
Any idea how we can filter out only tf events? i thought v2 is the only good indicator

Reply:
Example:
```
/mnt/logs/warp/few_rte-init-1614794261/log/train/:
total 72K
drwxr-xr-x 4 mahnerak mahnerak 4.0K Mar  3  2021 ..
drwxr-xr-x 2 mahnerak mahnerak 4.0K Mar  3  2021 .
-rw-r--r-- 1 mahnerak mahnerak  61K Mar  3  2021 events.out.tfevents.1614794275.4c0f0e71cd2f

/mnt/logs/warp/few_rte-init-1614794548/log/train/:
total 72K
drwxr-xr-x 4 mahnerak mahnerak 4.0K Mar  3  2021 ..
drwxr-xr-x 2 mahnerak mahnerak 4.0K Mar  3  2021 .
-rw-r--r-- 1 mahnerak mahnerak  61K Mar  3  2021 events.out.tfevents.1614794561.4c0f0e71cd2f

/mnt/logs/warp/few_rte-init-1614794842/log/train/:
total 72K
drwxr-xr-x 4 mahnerak mahnerak 4.0K Mar  3  2021 ..
drwxr-xr-x 2 mahnerak mahnerak 4.0K Mar  3  2021 .
-rw-r--r-- 1 mahnerak mahnerak  61K Mar  3  2021 events.out.tfevents.1614794860.4c0f0e71cd2f

/mnt/logs/warp/few_rte-init-1614795145/log/train/:
total 72K
drwxr-xr-x 4 mahnerak mahnerak 4.0K Mar  3  2021 ..
drwxr-xr-x 2 mahnerak mahnerak 4.0K Mar  3  2021 .
-rw-r--r-- 1 mahnerak mahnerak  61K Mar  3  2021 events.out.tfevents.1614795159.4c0f0e71cd2f
```

Reply:
will stick to `events.out.tfevents` pattern